Dockerfiles/midstream/unit_tests.py

```diff
@@ -0,0 +1,59 @@
+
+import unittest
+import os
+import subprocess
+
+
+class Testing(unittest.TestCase):
+    # Running the container with correctly set environment variables, all tasks from run_tests.py should pass
+    def test_postive(self):
+        exec_cmd = """
+        /run_tests.py
+        """
+        result = subprocess.run(exec_cmd, shell=True)
+        self.assertEqual(0, result.returncode)
+
+    # Set incorrect env variable UMOCI_BIN_PATH for ansible playbook extract-operator-bundle.yml - playbook fails to extract
+    def test_negative_extract_operator_bundle(self):
+        exec_cmd = """
+        export UMOCI_BIN_PATH=/root
+        /run_tests.py
+        """
+        result = subprocess.run(exec_cmd, shell=True)
+        with open("extract-operator-bundle.errormessage") as error_file:
+            if "Result code:" not in error_file.read():
+                print("Result code not found in %s" % error_file)
+                assert False
+        self.assertEqual(102, result.returncode)
+        
+    # Set incorrect env variable WORK_DIR for ansible playbook validate-operator-bundle.yml - playbook fails to validate
+    def test_negative_validate_operator_bundle(self):
+        exec_cmd = """
+        export WORK_DIR=/non_existing_dir
+        /run_tests.py
+        """
+        result = subprocess.run(exec_cmd, shell=True)
+        with open("validate-operator-bundle.errormessage") as error_file:
+            if "Result code:" not in error_file.read():
+                print("Result code not found in %s" % error_file)
+                assert False
+        self.assertEqual(103, result.returncode)
+
+    # Set env variable to custom made image that makes validation fails with following error:
+    # ERRO[0003] error validating format in /tmp/bundle-688328851: Bundle validation errors: couldn't
+    # parse dependency of type olm.crd
+    def test_negative_image_bundle_validation(self):
+        exec_cmd = """
+        export IMAGE_TO_TEST=quay.io/cvpops/test-operator:invalid-dependencies-v1
+        /run_tests.py
+        """
+        result = subprocess.run(exec_cmd, shell=True)
+        with open("/project/output/validation-output.txt") as error_file:
+            if "Bundle validation errors: couldn't parse dependency of type olm.crd" not in error_file.read():
```

Comment:
Can we change if to assert ? (self.assert.*)

Reply:
Will use self.assertTrue(), I would say it is the best fit here.

Reply:
There is also assertIn
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn